### PR TITLE
Create a stub for Azure Service Bus client

### PR DESCRIPTION
**Pull request by lgonczar**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RPE-70

### Change description ###

Create a stub for Azure Service Bus client. The stub allows for emulating input from Azure Service Bus subscription and will be used by tests.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/pbis/servicebus/ServiceBusClientStub.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    @Override
+    public IMessage receiveMessage() {
+        return messagesToReceive.poll();
+    }
+
+    @Override
+    public void completeMessage(String messageId, UUID messageLockToken) {
+        // nothing to be done
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/pbis/servicebus/ServiceBusClientStub.java]: how about we remove here and `peek` in method above?

File: src/test/java/uk/gov/hmcts/reform/pbis/servicebus/ServiceBusClientStubTest.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        clientStub.setMessagesToReceive(new LinkedList<>());
+        assertThat(clientStub.receiveMessage()).isNull();
+
+        clientStub.setMessagesToReceive(new LinkedList<>(nonEmptyList));
+        assertThat(clientStub.receiveMessage()).isNotNull();
+    }
+
+    @Test
+    public void completeMessage_should_not_throw_exception() {
+        clientStub.completeMessage("message-id-123", UUID.randomUUID());
```
Comment (rkondratowicz) [inline src/test/java/uk/gov/hmcts/reform/pbis/servicebus/ServiceBusClientStubTest.java]: There's:
```
assertThatCode(() -> {
  // code goes here
}).doesNotThrowAnyException();
```
in AssertJ, which is nice imho.

Up to you.

File: src/main/java/uk/gov/hmcts/reform/pbis/servicebus/ServiceBusClientStub.java (hunk 1 above)
Comment (lgonczar) [reply src/main/java/uk/gov/hmcts/reform/pbis/servicebus/ServiceBusClientStub.java]: Normally `receive()` gets the next item from the queue each time and `complete()` prevents it from coming back to the queue when the lock expires. I think the stub is closer to the real thing when it removes messages from the queue in `receive()`.

File: src/test/java/uk/gov/hmcts/reform/pbis/servicebus/ServiceBusClientStubTest.java (hunk 2 above)
Comment (lgonczar) [reply src/test/java/uk/gov/hmcts/reform/pbis/servicebus/ServiceBusClientStubTest.java]: Good idea, it's very explicit. I'll change that.

**Pull request merged**